src/shapes/textbox.class.ts

```diff
@@ -293,6 +304,83 @@ export class Textbox extends IText {
     return value.split(this._wordJoiners);
   }
 
+  /**
+   * Wraps a line of text using the width of the Textbox and a context. For overflowBreakWord option
+   * @param {Array} line The grapheme array that represent the line
+   * @param {Number} lineIndex
+   * @param {Number} desiredWidth width you want to wrap the line to
+   * @param {Number} reservedSpace space to remove from wrapping for custom functionalities
+   * @returns {Array} Array of line(s) into which the given text is wrapped
+   * to.
+   */
+  _wrapLineOfWordBreak(
+    _line,
+    lineIndex: number,
+    desiredWidth: number,
+    reservedSpace = 0
+  ): Array<any> {
+    desiredWidth -= reservedSpace;
+    const graphemeLines = [];
+    let text = _line || '',
+      i = 0,
+      j = 0,
+      offset = 0,
+      temp = '',
+      width = 0,
+      prevGrapheme = '',
+      largestLetterWidth = 0;
+
+    if (!text.length) {
+      graphemeLines.push([]);
+    }
+    while (text.length > 0) {
+      i = 0;
+      // Get the maximum string at a fixed width
+      width = 0;
+      prevGrapheme = '';
+      for (let k = 0, len = text.length; k < len; k++) {
+        const box = this._getGraphemeBox(
+          text[k],
+          lineIndex,
+          offset + k,
+          prevGrapheme,
+          true
+        );
+        width += box.kernedWidth;
+        largestLetterWidth = Math.max(largestLetterWidth, box.kernedWidth);
+        if (width > Math.max(largestLetterWidth, desiredWidth)) {
+          i = k;
+          break;
+        }
+        prevGrapheme = text[k];
+      }
+
+      // Prevent some edge case, may lead to an infinite loop
+      if (i === 0) {
+        graphemeLines.push(this.graphemeSplit(text));
+        break;
+      }
+
+      temp = text.substring(0, i);
+
+      if (text.length !== i) {
+        // Get last space to split words
+        const lastSpace = temp.lastIndexOf(' ');
```

Comment:
we should use a regex (`this._wordJoiners`) here in case of tabs and stuff I think.
Thoughts?

Reply:
Since `lastIndexOf` only supports string, I changed the implementation

Reply:
looks fine
There are other options, not sure if it makes any difference, things such as `String#matchAll` iterator or a more complex regex:
last occurrence of `\s`: https://regex101.com/r/DVP2Ix/1
all up to last occurrence of `\s`: https://regex101.com/r/3cwB7r/1

Reply:
We could then use `this._wordJoiners.source` to compose a new one.
But again, no idea if this impacts perf, and if it does it is probably extremely minor

Reply:
> looks fine There are other options, not sure if it makes any difference, things such as `String#matchAll` iterator or a more complex regex: last occurrence of `\s`: https://regex101.com/r/DVP2Ix/1 all up to last occurrence of `\s`: https://regex101.com/r/3cwB7r/1

I wrote some simple tests, using the regular in the example takes far more time than a simple loop

Reply:
Nice